src/transformers/modeling_tapas_utilities.py

```diff
@@ -0,0 +1,855 @@
+# coding=utf-8
```

Comment:
Same comment for the tokenizer utilities.

Reply:
Ok, where is the best place to add everything of `modeling_tapas_utilities.py`: at the top of `modeling_tapas.py`, or before/after the forward() of `TapasForQuestionAnswering`? (more general question: does one always need to define functions before using them, or does this not matter?)

And I don't understand what you mean by the wrapper, you mean I have to use that wrapper around every torch.function in my `modeling_tapas.py` file? Do you want to make PyTorch a soft dependency?

Reply:
I don't think we have a standard for where to put this, but as a general code readability observation I guess it would make sense to put the most important methods/classes higher up in the file, making them accessible quicker. Then it would mean putting the utilities at the bottom of the file, after the classes. What do you think, and what's your opinion on this @sshleifer, @patrickvonplaten, @sgugger?

No, I'm sorry I wasn't clear enough, I meant to say the `scatter` module, not PyTorch itself.

Reply:
I agree with @LysandreJik, it would be awesome to have all this functionality in `modeling_tapas.py`. I would put these functions at the bottom of the file.